pages/api/definitions/[version]/[document].js

```diff
@@ -0,0 +1,25 @@
+/**
+ * This API handler is for our JSON Schema files for validating AsyncAPI documents. 
+ * 
+ * It is an extension of the repository https://github.com/asyncapi/spec-json-schemas
+ * 
+ * Because we need to disable the normal rendering of pages, we rewrite any incoming requests 
+ * that is made to the path `/definitions` and reroute it (internally) to `/api/definitions`.
+ * 
+ * This is done in the the next.config.js file in the root folder.
+ * 
+ * @param {*} req 
+ * @param {*} res 
+ */
+export default function handler(req, res) {
+  const { document, version } = req.query
+  try {
+    // eslint-disable-next-line no-undef
+    const schema = require(`@asyncapi/specs/definitions/${version}/${document}`)
```

Comment:
I don't think this is going to work. Our website is not running server-side code but only compiled/bundled HTML+CSS+JS. Not sure if Netlify offers the possibility to have server-side code either.

Reply:
hmmmm, gooood question, https://deploy-preview-502--asyncapi-website.netlify.app/definitions/2.2.0/asyncapi.json certainly does not work.

Reply:
Think I have a workaround by utilizing the build scripts.

Reply:
Yup, that's the only way. We do everything at build time.

Reply:
It is honestly also the safest solution 😄

Reply:
> Yup, that's the only way. We do everything at build time.

> It is honestly also the safest solution 😄

Remember we need then to trigger a new build every time we merge a PR that changes the schemas at https://github.com/asyncapi/spec-json-schemas

Reply:
Will be done automatically, because our CI system update the package versions and create a new release 🙂

Reply:
That's right 👍 

BTW, there is another way for making this happen. Using Netlify rewrites so URL's can be redirected (proxied) to GH raw files, meaning we won't need to care about compiling the schemas here but just Netlify will grab the content from GH raw files (using caching mechanisms provided by GH CDN). It is as easy as this example shows https://docs.netlify.com/routing/redirects/rewrites-proxies/#proxy-to-another-service

The cons is that if we ever change the branch name, directory names or shape, those links will be broken 😓 .
Just to let you know it is a thing as well!